weaver/core/drivers/fabric-driver/server/satp.ts

```diff
@@ -0,0 +1,363 @@
+    createAssetRequest2['maturitydate'] = '05 May 48 00:00 MST';
+    createAssetRequest2['channel'] = 'mychannel';
+    createAssetRequest2['chaincode-id'] = 'satpsimpleasset';
+
+    const targetNetwork = createAssetRequest2['target-network'];
+    const owner = createAssetRequest2['owner'];
+    const ccType = createAssetRequest2['type'];
+    const assetType = createAssetRequest2['assetType'];
+    const id = createAssetRequest2['id'];
+    const issuer = createAssetRequest2['issuer'];
+    const facevalue = createAssetRequest2['facevalue'];
+    const maturitydate = createAssetRequest2['maturitydate'];
+    const tokenassettype = createAssetRequest2['tokenassettype'];
+    const numunits = createAssetRequest2['numunits'];
+    const channel = createAssetRequest2['channel'];
+    const chaincodeId = createAssetRequest2['chaincode-id'];
+
+    let gateway: Gateway = await getNetworkGateway(networkName);
+    const network: Network = await gateway.getNetwork(channel);
+    const contract = network.getContract(chaincodeId);
+    const currentQuery = {
+        channel: channel,
+        contractName: chaincodeId,
+        ccFunc: '',
+        args: []
+    }
+
+    const driverkeyCert = await getDriverKeyCert();
+    const certificate = Buffer.from(driverkeyCert.cert).toString('base64')
+
+    if (ccType == 'bond') {
+        currentQuery.ccFunc = 'CreateAsset'
+        currentQuery.args = [...currentQuery.args, assetType, id, certificate, issuer, facevalue, maturitydate]
+    } else if (ccType == 'token') {
+        currentQuery.ccFunc = 'IssueTokenAssets'
+        currentQuery.args = [...currentQuery.args, tokenassettype, numunits, certificate]
+    } else {
+        throw new Error(`Unrecognized asset category: ${ccType}`)
+    }
+    console.log(currentQuery)
+
+    try {
+        console.info(`Trying creating the asset: type: ${ccType}, id: ${id}, by: ${owner}, facevalue: ${facevalue}, maturitydate: ${maturitydate}`)
+        const read = await contract.submitTransaction(currentQuery.ccFunc, ...currentQuery.args)
+        const state = Buffer.from(read).toString()
+        if (state) {
+            logger.debug(`Response From Network: ${state}`)
+            console.info('Asset has been created successfully')
+        } else {
+            logger.debug('No Response from network')
+        }
+    } catch (error) {
+        console.error(`Failed to submit transaction: ${error}`)
+        throw new Error(error)
+    }
+
+    await gateway.disconnect()
+    logger.info('Gateways disconnected.')
+
+    const client = getRelayClientForAssetStatusResponse();
+    const request = new satp_pb.SendAssetStatusRequest();
+    request.setSessionId(createAssetRequest.getSessionId());
+    request.setStatus("Created");
+    client.sendAssetStatus(request, relayCallback);
+}
+
+async function extinguishHelper(
+    extinguishRequest: driverPb.ExtinguishRequest,
+    networkName: string
+): Promise<any> {
+
+    // TODO
```

Comment:
What does this TODO mean? Also fix the return type

Reply:
@outsidethecode Please replace with `unknown` as per the earlier suggestion. Also add `run the appropriate extinguish logic` after the `// TODO`.

Reply:
Fixed.